targets/mbedos5/jerryscript-mbed/jerryscript-mbed-drivers/source/I2C.cpp

```diff
@@ -0,0 +1,271 @@
+    I2C* native_ptr = reinterpret_cast<I2C*>(native_handle);
+
+    int hz = jerry_get_number_value(args[0]);
+    native_ptr->frequency(hz);
+
+    return jerry_create_undefined();
+}
+
+/**
+ * I2C#read (native JavaScript method)
+ *
+ * Read data from the I2C bus.
+ *
+ * @overload I2C#read(int)
+ * Read a single byte from the I2C bus
+ *
+ * @param ack indicates if the byte is to be acknowledged (1 => acknowledge)
+ *
+ * @returns array: Data read from the I2C bus
+ *
+ * @overload I2C#read(int, array, int)
+ * Read a series of bytes from the I2C bus
+ *
+ * @param address I2C address to read from
+ * @param data Array to read into
+ * @param length Length of data to read
+ *
+ * @returns array: Data read from the I2C bus
+ */
+DECLARE_CLASS_FUNCTION(I2C, read) {
+    if (args_count == 1) {
+        CHECK_ARGUMENT_TYPE_ALWAYS(I2C, read, 0, number);
+        uintptr_t native_handle;
+        jerry_get_object_native_handle(this_obj, &native_handle);
+        I2C *native_ptr = reinterpret_cast<I2C*>(native_handle);
+
+        int data = jerry_get_number_value(args[0]);
+        int result = native_ptr->read(data);
+
+        return jerry_create_number(result);
+    } else if (args_count == 3 || args_count == 4) {
+        CHECK_ARGUMENT_TYPE_ALWAYS(I2C, read, 0, number);
+        CHECK_ARGUMENT_TYPE_ALWAYS(I2C, read, 1, array);
+        CHECK_ARGUMENT_TYPE_ALWAYS(I2C, read, 2, number);
+
+        CHECK_ARGUMENT_TYPE_ON_CONDITION(I2C, read, 3, boolean, (args_count == 4));
+
+        uintptr_t native_handle;
+        jerry_get_object_native_handle(this_obj, &native_handle);
+        I2C *native_ptr = reinterpret_cast<I2C*>(native_handle);
+
+        const uint32_t data_len = jerry_get_array_length(args[1]);
+
+        int address = jerry_get_number_value(args[0]);
+        int length = jerry_get_number_value(args[2]);
+
+        char *data = new char[data_len];
+
+        bool repeated = false;
+        if (args_count == 4) {
+            repeated = jerry_get_boolean_value(args[3]);
+        }
+
+        int result = native_ptr->read(address, data, length, repeated);
+
+        jerry_value_t out_array = jerry_create_array(data_len);
+
+        for (uint32_t i = 0; i < data_len; i++) {
+            jerry_value_t val = jerry_create_number(double(data[i]));
+            jerry_set_property_by_index(out_array, i, val);
+            jerry_release_value(val);
+        }
+
+        delete[] data;
+
+        if (result == 0) {
+            // ACK
+            return out_array;
+        } else {
+            // NACK
+            const char *error_msg = "NACK received from I2C bus";
+
+            jerry_release_value(out_array);
+            return jerry_create_error(JERRY_ERROR_COMMON, reinterpret_cast<const jerry_char_t *>(error_msg));
+        }
+    }
+
+    const char* error_msg = "ERROR: Unexpected number of arguments to I2C.read, expected 1, 3 or 4.";
+    return jerry_create_error(JERRY_ERROR_TYPE, reinterpret_cast<const jerry_char_t*>(error_msg));
```

Comment:
Couldn't this be replaced with `CHECK_ARGUMENT_COUNT(I2C, read, (args_count == 1 || args_count == 3 || args_count == 4));`, moved to the start of the function?